internal/cmd/cli/apply/apply.go

```diff
@@ -177,32 +197,55 @@ func CreateBundles(ctx context.Context, client client.Client, repoName string, b
 // separated by a character set in opts.
 // If no fleet file is provided it tries to load a fleet.yaml in the root of the dir, or will consider
 // the directory as a raw content folder.
-func CreateBundlesDriven(ctx context.Context, client client.Client, repoName string, baseDirs []string, opts Options) error {
+func CreateBundlesDriven(pctx context.Context, client client.Client, repoName string, baseDirs []string, opts Options) error {
 	if len(baseDirs) == 0 {
 		baseDirs = []string{"."}
 	}
 
-	foundBundle := false
-	gitRepoBundlesMap := make(map[string]bool)
+	eg, ctx := errgroup.WithContext(pctx)
+	eg.SetLimit(bundleCreationMaxConcurrency)
+	bundlesChan := make(chan *fleet.Bundle)
 	for _, baseDir := range baseDirs {
 		opts := opts
-		// verify if it also defines a fleetFile
-		var err error
-		baseDir, opts.BundleFile, err = getPathAndFleetYaml(baseDir, opts.DrivenScanSeparator)
-		if err != nil {
-			return err
-		}
-		if auth, ok := opts.AuthByPath[baseDir]; ok {
-			opts.Auth = auth
-		}
-		if err := Dir(ctx, client, repoName, baseDir, &opts, gitRepoBundlesMap); err == ErrNoResources {
-			logrus.Warnf("%s: %v", baseDir, err)
-			return nil
-		} else if err != nil {
-			return err
-		}
-		foundBundle = true
+		eg.Go(func() error {
+			// verify if it also defines a fleetFile
+			var err error
+			baseDir, opts.BundleFile, err = getPathAndFleetYaml(baseDir, opts.DrivenScanSeparator)
+			if err != nil {
+				return err
+			}
+			if auth, ok := opts.AuthByPath[baseDir]; ok {
+				opts.Auth = auth
+			}
+			bundle, scans, err := Dir(ctx, repoName, baseDir, opts)
+			if err != nil {
+				if err == ErrNoResources {
+					logrus.Warnf("%s: %v", baseDir, err)
+					return nil
+				}
+				return err
+			}
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			case bundlesChan <- bundle:
+			}
+			return writeBundle(ctx, client, bundle, scans, opts)
+		})
 	}
+	go func() {
+		_ = eg.Wait()
+		close(bundlesChan)
```

Comment:
Couldn't we just return in here if there's any error? 
The second `eg.Wait()` would not be required, then. 

Does the second `Wait` really wait? (reaching that code depends on closing the channel, which is done when the first `Wait` finished)